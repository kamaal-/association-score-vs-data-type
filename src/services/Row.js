import {initialDataTypes} from '../config/'

export default class Row {
    constructor(data){
        this.data = data 
        this.setupDataTypes()
    }

    setupDataTypes(data){
        const self = this
        this.dataTypeKeyValues = (Object.keys(initialDataTypes)|| []).map( key => {
            return {...initialDataTypes[key], value: self.data.association_score.datatypes[key]}
        })
    }

    get geneId(){
        return this.data.target.id
    }

    get symbol(){
        return this.data.target.gene_info.symbol
    }

    get name(){
        return this.data.target.gene_info.name
    }

    get overallScore(){
        return this.data.association_score.overall
    }

    get visualizerData(){
        return this.dataTypeKeyValues
    }

    set dataType(dataType){
        this.dataTypeKeyValues.push(dataType)
    }
}