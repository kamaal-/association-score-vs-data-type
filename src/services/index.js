import Axios from "axios"
import Row from './Row'
import {totalItemsFromTo} from '../config/'

export const dataToRow = d => new Row(d)

export const transformDataToRows  = data => ((data && data.data) || [])
    .map(dataToRow)
    .sort(sortRowByOverallScore)
    .slice(...totalItemsFromTo)

export const sortRowByOverallScore = (a, b) => {
    if(a.overallScore > b.overallScore) return -1;
    if(a.overallScore < b.overallScore) return 1;
    return 0;
}

export const getInitialData = (cb) => {
    Axios.get('https://demo6922545.mockable.io/')
        .then(({data}) => cb(transformDataToRows (data)))
        .catch(e => {
            console.log(e)
        })
}