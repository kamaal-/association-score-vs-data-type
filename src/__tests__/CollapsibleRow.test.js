import React from 'react';
import { render } from '@testing-library/react';
import Row from '../components/Rows/CollapsibleRow';
import data from '../config/mock.data';
import { transformDataToRows Â } from '../services/'
import ReactDOM from 'react-dom'

describe('Collapsible Row', () => {

    const setup = () => {
        const container = document.createElement('tbody')
        const rows = transformDataToRows(data())	
        return render(<Row row={rows[0]} index={0} collapsedRow={2} chartTitle={'Chart Title'}/>, {
            container: document.body.appendChild(container)
        });
    }

  	it('renders without crashing', () => {
        const div = document.createElement('tbody');
        const rows = transformDataToRows(data())	
    	ReactDOM.render(<Row row={rows[0]} index={0} collapsedRow={2} chartTitle={'Chart Title'}/>, div);
    	ReactDOM.unmountComponentAtNode(div);
    });

  	it('Should render 1 rows', () => {
		const { getAllByTestId } = setup()
		expect( getAllByTestId('collapsible-tr').length).toBe(1);
    });
      
    it('should match snapshot', () => {
        const { asFragment } = setup()
        expect(asFragment()).toMatchSnapshot();
    })

})
 