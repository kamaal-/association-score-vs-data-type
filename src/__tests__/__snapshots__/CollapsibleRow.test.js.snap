// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible Row should match snapshot 1`] = `
<DocumentFragment>
  <tr
    class="MuiTableRow-root"
    data-testid="collapsible-tr"
  >
    <td
      class="MuiTableCell-root"
      colspan="5"
      style="padding-bottom: 0px; padding-top: 0px;"
    />
  </tr>
</DocumentFragment>
`;
